client/components/ResponsiveFlex/ResponsiveFlex.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { PropsWithChildren, useEffect } from 'react';
+
+interface Props {
+  breakpoint: number;
+  className?: string;
+}
+
+// This is a generic component that receives the breakpoint and a className as props and
+// then switch from row to column layout.
+// Because this component will be used in multiple places, we wanted a simple implementation,
+// with very little js.
+
+function ResponsiveFlex({ breakpoint, children, className = '' } : PropsWithChildren<Props>) {
+  const containerRef = React.createRef<HTMLDivElement>();
+  const responsiveClassName = `responsive-flex--${breakpoint}`;
+
+  // we first check if we have a unique class on the container and if we don't,
```

Comment:
Add this to the begining:

`For each breakpoint value we create a className with the rules specific to that breakpoint`.
 
I think it provides more clarity. Then you can continue with your explanation 👍  (we first check if the class already exists. If not ...)